backend/agentpress/response_processor.py

```
@@ -400,30 +415,33 @@ async def process_streaming_response(
+                            if execution["task"].done():
+                                result = execution["task"].result()
+                                context.result = result
+                                tool_results_buffer.append((execution["tool_call"], result, tool_idx, context))
+                                
+                                if tool_name in ['ask', 'complete']:
+                                    logger.info(f"Terminating tool '{tool_name}' completed during streaming. Setting termination flag.")
+                                    self.trace.event(name="terminating_tool_completed_during_streaming", level="DEFAULT", status_message=f"Terminating tool '{tool_name}' completed during streaming. Setting termination flag.")
+                                    record_event(name="terminating_tool_completed_during_streaming", level="DEFAULT", message=f"Terminating tool '{tool_name}' completed during streaming. Setting termination flag.")
+                                    agent_should_terminate = True
+                                    
+                            else: # Should not happen with asyncio.wait
```

Flake8 (7.2.0)

[error] at least two spaces before inline comment

(E261)